lib/matplotlib/axes/_axes.py

```diff
@@ -908,6 +908,79 @@ def axvline(self, x=0, ymin=0, ymax=1, **kwargs):
         self.autoscale_view(scalex=scalex, scaley=False)
         return l
 
+    @docstring.dedent_interpd
+    def axline(self, xy1, xy2, **kwargs):
+        """
+        Add an infinitely long straight line that passes through two points.
+
+        Parameters
+        ----------
+        xy1, xy2 : (float, float)
+            Points for the line to pass through.
+
+        Returns
+        -------
+        :class:`~matplotlib.lines.Line2D`
+
+        Other Parameters
+        ----------------
+        Valid kwargs are :class:`~matplotlib.lines.Line2D` properties,
+        with the exception of 'transform':
+
+        %(_Line2D_docstr)s
+
+        Examples
+        --------
+        * Draw a thick red line passing through (0, 0) with a gradient of 1::
```

Comment:
```suggestion
        Draw a thick red line passing through (0, 0) an (1, 1)::
```

- No need for a bullet list with one item.
- Stick closer to the two-point semantic of the API. People who want to use gradients will know hot to get their gradient transformed to a second point. For others this is rather confusing as the gradient 1 is only implicitly defined by the points in the example.